docs/src/components/FlutterAuthenticatorExample.tsx

```diff
@@ -12,15 +14,20 @@ export function FlutterAuthenticatorExample({
   // it is used by the authenticator to signal when the authenticator has finished loading.
   id = generateId(),
 }) {
-  const colorAttr = 'data-amplify-color-mode';
-  // in dev mode, `data-amplify-color-mode` will not be on the html element
-  const colorElements =
-    typeof window !== 'undefined'
-      ? document.querySelectorAll(`[${colorAttr}]`)
-      : [];
-  const themeMode = colorElements.length
-    ? colorElements[0].getAttribute(colorAttr)
-    : 'light';
+  const [themeMode, setThemeMode] = React.useState('light');
+  React.useLayoutEffect(() => {
+    let themeMode = document.documentElement.getAttribute(
+      'data-amplify-color-mode'
+    );
+    if (themeMode === 'system') {
+      if (window.matchMedia('(prefers-color-scheme:dark)').matches) {
+        themeMode = 'dark';
+      } else {
+        themeMode = 'light';
+      }
+    }
+    setThemeMode(themeMode);
+  }, []);
```

Comment:
Without full context might be missing something but it looks `lightMode` could be resolved sync. Is there not a hook that already provides this information though?

Reply:
I am not aware of an existing hook.

I have updated this because the original implementation wasn't working as expected, but I am wondering if there is a better way to do this.

The current implementation in main is flawed because `data-amplify-color-mode` is not set when this code runs on a page refresh. That results in themeMode always being "light" after a page refresh.

I initially used `useLayoutEffect` to wait until DOM mutations are complete. This also is flawed because themeMode never updates.

The current change in the PR uses `useLayoutEffect` but passes in a fn that returns the value of `getAttribute('data-amplify-color-mode')` as a dependency. This forces the fn to run when that attribute updates. I am not sure if there is a better way to do in react.

Reply:
I could update the PR to change `useLayoutEffect` to `useEffect`. `useLayoutEffect` is not needed now that the getter for data-amplify-color-mode is a dependency.

Reply:
I reverted these changes as the issue they aimed to fix were not related to this PR and it doesn't seem to be reproducible in prod.